notes/stv/20191120-openstack-deployment.txt

```diff
@@ -0,0 +1,761 @@
+10.0.0.10       stv-aglais-worker-3
+10.0.0.5      stv-aglais-zeppelin
+
+EOF
+  
+exit
+
+
+cat > "${HOME:?}/.ssh/config" << EOF
+
+    Host stv-aglais-worker-1
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-worker-2 
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-worker-3
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-master.novalocal
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-master
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-zeppelin
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+EOF
+
+
+sudo chmod 600 ~/.ssh/config
+
+
+
+mkdir /home/fedora/spark
+
+## Copy Hadoop & Spark Config files from master
+ssh stv-aglais-master
+    scp ~/hadoop/etc/hadoop/* stv-aglais-master:/home/fedora/zeppelin/hadoop-conf;
+    scp -r ~/spark/* stv-aglais-zeppelin:/home/fedora/spark;
+exit
+
+
+
+## Enable Python3 and matplotlib for python3 notebooks
+## Update python interpreter, set it to python3
+
+## Install matplotlib
+sudo python3 -m pip install -U pip
+sudo python3 -m pip install -U matplotlib
+sudo python3 -m pip install -U healpy
+
+
+
+## Start Server
+/home/fedora/zeppelin/bin/zeppelin-daemon.sh start
+
+
+## Tunnel connection to 8080 and go to
+## On Local Machine (Browser)
+## Go to: http://localhost:8080/#/
+
+
+
+## Update Spark Interpreter from admin GUI
+## Set master=yarn-client
+
+
+
+
+#-------------------------------------------------------------------------
+#  Gaiasource & Tgas as Parquet files
+#-------------------------------------------------------------------------
+
+## From Gaia-spark virtual image (GDAF), copy ascii, parquet files from /archive/gaia/gdr1 to /hadoop/gaia/parquet & /hadoop/gaia/parquet on master node
+## If we want to use the Spark cluster to generate the parquet files from the ascii ones by running the asciiToParquetConverter jar through spark, we'd have to copy
+## the ascii files on each worker node?
+
+## To convert, run:
+## Run from directory where asciiToParquetConverter.jar is
+
+    spark-submit --num-executors 3 --driver-memory 4096m --executor-memory 1024m --executor-cores 2 asciiToParquetConverter.jar -input file:///hadoop/gaia/ascii/gaia/data -output file:///hadoop/gaia/parquet/gaia/gdr1/gaiaSource/ -schema file:///hadoop/gaia/ascii/gaia/GaiaSourceHeaderSchema
+  
+
+
+## Put parquet files on HDFS:
```

Comment:
If the spark job results were output to `file:///hadoop/gaia/parquet/gaia/gdr1/gaiaSource/`, then why do we need this step ?